libkmod/libkmod-internal.h

```diff
@@ -68,16 +68,16 @@ enum kmod_file_compression_type {
 	KMOD_FILE_COMPRESSION_ZLIB,
 };
 
-struct kmod_list *kmod_list_append(struct kmod_list *list, const void *data) _must_check_ __attribute__((nonnull(2)));
-struct kmod_list *kmod_list_prepend(struct kmod_list *list, const void *data) _must_check_ __attribute__((nonnull(2)));
```

Comment:
I don't understand why you think this one is ok. The _must_check_ is serving its purpose that nobody should call this function without checking the return.

Reply:
My proposal is that we drop that patch, just replacing with one that moves it to the left. See branch ci-test - 
 up to 0c8f1271eda6 ("Remove -Wno-unused-result")